apps/src/lab2/responseValidators.ts

```diff
@@ -52,3 +80,32 @@ export class ValidationError extends Error {
     Object.setPrototypeOf(this, ValidationError.prototype);
   }
 }
+
+export function setValidatorForAppType(appName: AppName) {
+  if (appName === 'pythonlab') {
+    Lab2Registry.getInstance().setSourceResponseValidator(
+      PythonSourceResponseValidator
+    );
+  } else if (BLOCKLY_LABS.includes(appName)) {
+    // Blockly labs
+    Lab2Registry.getInstance().setSourceResponseValidator(
+      BlocklySourceResponseValidator
+    );
+  } else {
+    // Everything else uses the default validator
+    Lab2Registry.getInstance().setSourceResponseValidator(
+      DefaultSourceResponseValidator
+    );
+  }
+}
+
+function sourceValidatorHelper(
+  response: Record<string, unknown>,
+  appSpecificValidator: (response: Record<string, unknown>) => void
+): ProjectSources {
+  if (!response.source) {
+    throw new ValidationError('Missing required field: source');
+  }
+  appSpecificValidator(response);
+  return response as unknown as ProjectSources;
```

Comment:
Why does this need to have `as unknown as`?

Reply:
This is how you force convert a type to another type that TypeScript thinks is not compatible. `unknown` is a version of `any` that can only be used with more type assertions [reference](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type). Since we are getting data from an outside source, we validate it, then cast it once we know it's valid.